webapp/login.go

```diff
@@ -16,6 +17,34 @@ import (
 	"golang.org/x/oauth2"
 )
 
+func loginStatusHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := shared.NewAppEngineContext(r)
+	aeAPI := shared.NewAppEngineAPI(ctx)
+	if !aeAPI.IsFeatureEnabled("githubLogin") {
+		http.Error(w, "Feature not enabled", http.StatusNotImplemented)
+		return
+	}
+
+	if cookie, err := r.Cookie("session"); err != nil || cookie == nil {
+		respond, _ := json.Marshal("User is not logged in")
```

Comment:
Hmm, this will produce `"User is not logged in"` (with the quotes), which is technically JSON but not really more ergonomic, especially when you look at the success case: callers of this API would have to parse that string to get the username. If GitHub allows spaces in usernames, this would be even more brittle.

I was thinking we could return a struct like
```go
type response struct {
  User *shared.User `json:,omitempty`
  Error string `json:,omitempty`
}
```

That way, it's very easy to check whether the user is logged in (based on which field is present) and extract information.

Reply:
Thanks for the suggestion and I agree with you here! Perhaps we can have `Error *string` instead, so the error message is omitted when the response is success.

Reply:
No we don't have to; empty strings are omitted.